src/main/kotlin/no/nav/familie/tilbake/service/dokumentbestilling/vedtak/VedtaksbrevFritekstValidator.kt

```diff
@@ -0,0 +1,126 @@
+                                                   vedtaksbrevstype: Vedtaksbrevstype) {
+        val maksTekstLengde = when (vedtaksbrevstype) {
+            ORDINÆR -> 4000
+            else -> 10000
+        }
+        if (vedtaksbrevsoppsummering.oppsummeringFritekst != null &&
+            vedtaksbrevsoppsummering.oppsummeringFritekst.length >= maksTekstLengde) {
+            throw Feil(message = "Oppsummeringstekst er for lang for behandling ${behandling.id}",
+                       frontendFeilmelding = "Oppsummeringstekst er for lang for behandling ${behandling.id}",
+                       httpStatus = HttpStatus.BAD_REQUEST)
+        }
+    }
+
+    private fun validerFritekstIFaktaAvsnitt(faktaFeilutbetaling: FaktaFeilutbetaling,
+                                             vedtaksbrevFritekstPerioder: List<Vedtaksbrevsperiode>,
+                                             avsnittMedPerioder: List<PeriodeMedTekstDto>) {
+        faktaFeilutbetaling.perioder.filter { Hendelsesundertype.ANNET_FRITEKST == it.hendelsesundertype }
+                .forEach { faktaFeilutbetalingsperiode ->
+                    val perioder = finnFritekstPerioder(vedtaksbrevFritekstPerioder,
+                                                        faktaFeilutbetalingsperiode.periode,
+                                                        Friteksttype.FAKTA)
+                    if (perioder.isEmpty()) {
+                        throw Feil(message = "Mangler fakta fritekst for alle fakta perioder",
+                                   frontendFeilmelding = "Mangler Fakta fritekst for alle fakta perioder",
+                                   httpStatus = HttpStatus.BAD_REQUEST)
+                    }
+                    // Hvis en av de periodene mangler fritekst
+                    val omsluttetPerioder = avsnittMedPerioder.filter {
+                        faktaFeilutbetalingsperiode.periode.omslutter(Periode(it.periode))
+                    }
+                    omsluttetPerioder.forEach {
+                        if (it.faktaAvsnitt.isNullOrBlank()) {
+                            throw Feil(message = "Mangler fakta fritekst for ${it.periode.fom}-${it.periode.tom}",
+                                       frontendFeilmelding = "Mangler Fakta fritekst for ${it.periode.fom}-${it.periode.tom}",
+                                       httpStatus = HttpStatus.BAD_REQUEST)
+                        }
+                    }
+                }
+    }
+
+    private fun validerFritekstISærligGrunnerAnnetAvsnitt(vilkårsvurdering: Vilkårsvurdering,
+                                                          vedtaksbrevFritekstPerioder: List<Vedtaksbrevsperiode>) {
+        vilkårsvurdering.perioder.filter {
+            it.aktsomhet?.vilkårsvurderingSærligeGrunner != null &&
+            it.aktsomhet.vilkårsvurderingSærligeGrunner
+                    .any { særligGrunn -> SærligGrunn.ANNET == særligGrunn.særligGrunn }
```

Comment:
Du trenger ikke null-sjekken
it.aktsomhet?.vilkårsvurderingSærligeGrunner?.any { særligGrunn -> SærligGrunn.ANNET == særligGrunn.særligGrunn }

Reply:
Der trenger jeg null sjekken, ellerså returnerer det Boolean?